src/main/java/net/plavcak/jenkins/plugins/scmskip/SCMSkipTools.java

```diff
@@ -93,16 +93,20 @@ private static boolean inspectChangeSet(List<ChangeLogSet<?>> changeLogSets, SCM
 
         ChangeLogSet.Entry matchedEntry = null;
 
+        boolean allSkipped = true;
+        
         for (Object entry : changeLogSet.getItems()) {
             if (entry instanceof ChangeLogSet.Entry && inspectChangeSetEntry((Entry) entry, matcher)) {
                 matchedEntry = (Entry) entry;
+            } else {
+                allSkipped = false;
                 break;
```

Comment:
@worldjoe With this break, aren't you checking the first commit instead of the last?

Reply:
@debugmaster I looked at it too, but I think it is correct.

The "if else" is not testing if there is some (or more) ci skip. It is testing if there is some commit WITHOUT ci skip. If there is any, the `allSkipped` is set to false and the loop does not need to continue.

I think we don't want to skip build if there is some commit without ci skip.

Reply:
I see. If all commits match the pattern, then skip run. Otherwise, don't skip.